monai/transforms/compose.py

```diff
@@ -358,3 +358,62 @@ def inverse(self, data):
         for o in reversed(applied_order):
             data = apply_transform(self.transforms[o].inverse, data, self.map_items, self.unpack_items, self.log_stats)
         return data
+
+
+class SomeOf(Compose):
```

Comment:
I wrote a longer comment on this in the PR, but I think SomeOf should be implemented similarly to OneOf / RandomOrder. The list of transforms needs to stay static for this to work in the general case and with other parts of monai that assume the transform list doesn't change.

You can instead write the indices of the transforms that were called into the metadata of the tensor on the forward pass, and then iterate over those indices in reverse order during the call to __inverse__, calling the inverse of the transform at the given index.

Reply:
Thanks for the suggestion, I'll take a look at those examples.

Reply:
The latest commit is my attempt to refactor it to mimick the RandomOrder design pattern

Reply:
This looks really good. It is create to have them all consistent!